### PR TITLE
Update details of dependency analysis in the design doc

### PR DESCRIPTION
**Pull request by typhoonzero**
Distinguish statement read or write type: model and table.

### CODE REVIEW

File: doc/design/sql_program_dependency_analyze.md [hunk 1]
```diff
@@ -191,6 +193,9 @@ func Analyze(program []ir.Statement) (*deps.Statement, error) {}
 ```
 
 **NOTE: we treat table and model as the same thing when constructing the graph.**
+**The actual table name in the graph is "Type.Name", because we may save the model in OSS storage**
```
Comment (Yancey0623) [inline doc/design/sql_program_dependency_analyze.md:207]: Maybe adding a receiver to identify the actual table name can be more clear:
``` golang
Type Table struct {
    type string
    name string
    Inputs *[]Statement
    Outputs *[]Statement
}

fund (t *Table) Name() string {
    return t.Type + "." + t.Name
}
```

**Review by Yancey0623 — DISMISSED** [review]
LGTM, please feel free to update the design.

File: doc/design/sql_program_dependency_analyze.md [hunk 2]
```diff
@@ -179,6 +179,8 @@ type Statement struct {
 }
 
 type Table struct {
+  // Type can be "table" or "model".
```
Comment (shendiaomo) [inline doc/design/sql_program_dependency_analyze.md:189]: How about defining `Type` as an enumeration?

**Review by tonyyang-svail — APPROVED** [review]
LGTM.

**Pull request merged**